tuf/ngclient/updater.py

```diff
@@ -362,9 +363,12 @@ def _update_root_symlink(self) -> None:
         linkname = os.path.join(self._dir, "root.json")
         version = self._trusted_set.root.version
         current = os.path.join("root_history", f"{version}.root.json")
-        with contextlib.suppress(FileNotFoundError):
-            os.remove(linkname)
-        os.symlink(current, linkname)
+
+        with open(os.path.join(self._dir, current + ".lck"), "wb") as f:
```

Comment:
I think using the version number here is potentially harmful (the lock file is now `<metadata_dir>/root_history/<version>.root.json.lck`) -- but we are actually trying to protect access to the *unversioned* symlink.


```suggestion
        with open(os.path.join(self._dir, ".root.json.lck"), "wb") as f:
```

(I also added the leading dot but that's just a suggestion)

Reply:
Sounds good. Let me change this.